src/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -1025,5 +1025,49 @@ public static void MiscellaneousTest()
             Thread.SpinWait(1);
             Thread.Yield();
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
```

Comment:
What's the behavior on Unix?  Let's have a test for that.  Ideally it would end up throwing PlatformNotSupportedException; what does it actually fail with?

Reply:
It was throwing platformNotSupportedException but it was wrapped around some other exception. Adding DonNotwrapExtensions flag will help in correcting this

Reply:
So what's the behavior exactly on Unix now with the latest changes? I see that it is not throwing PNSE anymore. What is the exception that it throws now?

Reply:
It throws System.ArgumentException with this message ``` Cannot bind to the target method because its signature or security transparency is not compatible with that of the delegate type``` 
on unix from here https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/Delegate.cs#L422

Reply:
That is not a very good exception message. If somebody sees it, they won't be able to figure out what's going on. I  think we should make this explicitly throw PNSE for Unix.

Reply:
Sure
```
Type type = Type.GetType("System.Security.Principal.WindowsPrincipal, System.Security.Principal.Windows", throwOnError: true);
Volatile.Write(ref s_getWindowsPrincipal, (Func<IPrincipal>)Delegate.CreateDelegate(typeof(Func<IPrincipal>), type, "GetDefaultInstance", ignoreCase: false, throwOnBindFailure: false));
if (s_getWindowsPrincipal == null)
    throw System.ArgumentException(SR.exceptionString);
```
what string should I pass in Exception string ?

Reply:
Same string as what you get when you try to do `new System.Security.Principal.WindowsIdentity` on Unix:

`Unhandled Exception: System.PlatformNotSupportedException: Windows Principal functionality is not supported on this platform.`

Also, I think it should be done before the `Volatile.Write`.

Reply:
We should also change the original string at some point to not mention security transparency

Reply:
Can we not fix the core problem here that the binding is apparently working differently between the platforms?

Reply:
> We should also change the original string at some point to not mention security transparency

Done: https://github.com/dotnet/coreclr/pull/18776/files

Reply:
> Can we not fix the core problem here that the binding is apparently working differently between the platforms?

`System.Security.Principal.Windows` has auto-generated stub on non-Windows. The method we are binding to does not exist in it. Is there a better way to deal with this than what we are doing here now?

Reply:
> Is there a better way to deal with this than what we are doing here now?

One option would be to put the new GetDefaultInstance method onto a different new internal type.  The auto-generation mechanism adds stubs for everything in the reference assembly, but it doesn't prevent you from having additional stuff.  So if we just put that method onto some other internal type we define and include in all builds (and similarly exclude from ILLink), then we can bind to it in all builds.